lib/ortho_functions.py

```diff
@@ -238,9 +204,381 @@
+                    ulx = gcp_dict[1][2]
+                    uly = gcp_dict[1][3]
+                    urx = gcp_dict[2][2]
+                    ury = gcp_dict[2][3]
+                    llx = gcp_dict[4][2]
+                    lly = gcp_dict[4][3]
+                    lrx = gcp_dict[3][2]
+                    lry = gcp_dict[3][3]
+                    xsize = gcp_dict[1][0] - gcp_dict[2][0]
+                    ysize = gcp_dict[1][1] - gcp_dict[4][1]
+
+                else:
+                    xsize = ds.RasterXSize
+                    ysize = ds.RasterYSize
+                    proj = ds.GetProjectionRef()
+                    gtf = ds.GetGeoTransform()
+                    ulx = gtf[0] + 0 * gtf[1] + 0 * gtf[2]
+                    uly = gtf[3] + 0 * gtf[4] + 0 * gtf[5]
+                    urx = gtf[0] + xsize * gtf[1] + 0 * gtf[2]
+                    ury = gtf[3] + xsize * gtf[4] + 0 * gtf[5]
+                    llx = gtf[0] + 0 * gtf[1] + ysize * gtf[2]
+                    lly = gtf[3] + 0 * gtf[4] + ysize * gtf[5]
+                    lrx = gtf[0] + xsize * gtf[1] + ysize * gtf[2]
+                    lry = gtf[3] + xsize * gtf[4] + ysize * gtf[5]
+                ds = None
+
+                ####  Create geometry objects
+                ul_geom = ogr.Geometry(ogr.wkbPoint)
+                ul_geom.AddPoint(ulx, uly)
+                ur_geom = ogr.Geometry(ogr.wkbPoint)
+                ur_geom.AddPoint(urx, ury)
+                lr_geom = ogr.Geometry(ogr.wkbPoint)
+                lr_geom.AddPoint(lrx, lry)
+                ll_geom = ogr.Geometry(ogr.wkbPoint)
+                ll_geom.AddPoint(llx, lly)
+
+                ring = ogr.Geometry(ogr.wkbLinearRing)
+                ring.AddPoint(ulx, uly)
+                ring.AddPoint(urx, ury)
+                ring.AddPoint(lrx, lry)
+                ring.AddPoint(llx, lly)
+                ring.AddPoint(ulx, uly)
+                image_geom = ogr.Geometry(ogr.wkbPolygon)
+                image_geom.AddGeometry(ring)
+                g_srs = srs_wgs84
+
+                #### Create source srs objects
+                s_srs = utils.osr_srs_preserve_axis_order(osr.SpatialReference(proj))
+                try:
+                    sg_ct = osr.CoordinateTransformation(s_srs, g_srs)
+                except RuntimeError as e:
+                    logger.error(f"Source image coordinate system error: {self.src_image} - {e}")
+                    rc = 1
+                else:
+                    #### Transform geometries to geographic
+                    if not s_srs.IsSame(g_srs):
+                        ul_geom.Transform(sg_ct)
+                        ur_geom.Transform(sg_ct)
+                        ll_geom.Transform(sg_ct)
+                        lr_geom.Transform(sg_ct)
+                        image_geom.Transform(sg_ct)
+                    logger.info("Geographic extent: %s", str(image_geom))
```

Comment:
Log fires on every scene. Consider changing to `logger.debug()` for opt-in verbosity.